app/controllers/concerns/current_structure.rb

```diff
@@ -0,0 +1,72 @@
+module CurrentStructure
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_structure_type_in_session, :set_session_organisation_id, :set_session_department_id,
+                  :set_current_structure_attributes
+
+    helper_method :structure_type, :department_level?
+  end
+
+  def set_structure_type_in_session
+    return if params[:department_id].nil? && params[:organisation_id].nil?
+
+    session[:structure_type] = params[:organisation_id].present? ? "organisation" : "department"
+  end
+
+  def set_session_organisation_id
+    if session[:structure_type] == "organisation"
+      session[:organisation_id] = params[:organisation_id] if params[:organisation_id]
+    else
+      session[:organisation_id] = nil
+    end
+  end
+
+  def set_session_department_id
+    if session[:structure_type] == "department"
+      session[:department_id] = params[:department_id] if params[:department_id]
+    else
+      session[:department_id] = nil
+    end
+  end
+
+  def set_current_structure_attributes
+    Current.structure_type = session[:structure_type]
+    Current.organisation_id = session[:organisation_id]
+    Current.department_id = session[:department_id]
+  end
+
+  def structure_type
+    Current.structure_type
+  end
```

Comment:
Not worth it IMO, ça laisse penser que structure_type a un behavior plus complexe qu'un simple passe plat

Reply:
C'était principalement pour éviter d'avoir à écrire `Current.structure_type` à chaque fois quand on définit nos routes mais ok 👍. 
Done here: https://github.com/betagouv/rdv-insertion/pull/1581/commits/9caa7ed0fab005f96bcaed65051bd4efa372ca69